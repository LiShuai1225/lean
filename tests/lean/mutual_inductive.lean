constant (f : nat -> nat -> Type)
constant (g : nat -> Type)
constant (f00 : f 0 0)

mutual_inductive foo₁, foo₂, foo₃, foo₄ (A : Type) (a : A)
with foo₁ : A -> Type
| mk₁ : foo₁ a
with foo₂ : nat -> Type
| mk₂ : forall (b : A) (n : nat), g n -> foo₁ b -> foo₂ (n+1)
with foo₃ : forall {n₁ n₂ : nat}, f n₁ n₂ -> Type
| mk₃ : A -> foo₂ 0 -> foo₃ f00
with foo₄ : Type
| mk₄ : foo₂ 0 → foo₄

set_option pp.binder_types true
check @foo₁
check @foo₂
check @foo₃
check @foo₄

check @foo₁.mk₁
check @foo₂.mk₂
check @foo₃.mk₃
check @foo₄.mk₄
