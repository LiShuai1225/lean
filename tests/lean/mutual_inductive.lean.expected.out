foo₁ : Π (A : Type), A → A → Type
foo₂ : Π (A : Type), A → ℕ → Type
foo₃ : Π (A : Type), A → (Π {n₁ n₂ : ℕ}, f n₁ n₂ → Type)
foo₄ : Π (A : Type), A → Type
foo₁.mk₁ : Π {A : Type} (a : A), foo₁ A a a
foo₂.mk₂ : Π {A : Type} {a : A} (b : A) (n : ℕ), g n → foo₁ A a b → foo₂ A a (n + 1)
foo₃.mk₃ : Π {A : Type} {a : A}, A → foo₂ A a 0 → foo₃ A a f00
foo₄.mk₄ : Π {A : Type} {a : A}, foo₂ A a 0 → foo₄ A a
